nucleus/parent/pom.xml

```diff
@@ -583,6 +583,11 @@
                 <scope>import</scope>
                 <type>pom</type>
             </dependency>
+            <dependency>
+                <groupId>org.glassfish.hk2</groupId>
+                <artifactId>hk2-config</artifactId>
+                <version>${project.version}</version>
```

Comment:
GlassFish does not manage dependencies of its own modules. Why do you need this change ?

Reply:
Why is hk2-config 5.1.0 still in the dependency graph, is this coming from jersey ? I thought you removed these.

This is lipstick on the pig, instead we should fix the dependency graph in GF where needed. (e.g. glassfish-api does has a compile time dependency on hk2-config but it is resolved transitively),.

Reply:
I have put it here as it solves it for many. Currently, GF 5.1.0 is removed from staging. Basically, there is no try right now if wanted to put away

Reply:
I don't understand what you are saying.

Even if GF 5.1.0 would still be available from the staging repository something has/had a dependency on it for you to make these changes. What brings it, and **why**.

Get rid of it.

Reply:
Jersey does not. You would have seen it in Jersey MANIFEST otherwise.

Reply:
If jersey does not depend on hk2-config:5.1.0 then what brings it in ?

Reply:
Hm...now we reverted back in Jersey to version 2.5.0. But it is no longer in gf, in gf is 5.0.1. So it fails with missing requirement [org.glassfish.main.common.glassfish-api [94](R 94.0)] osgi.wiring.package; (&(osgi.wiring.package=org.jvnet.hk2.config)(version>=2.5.0)(!(version>=3.0.0)))]]. 

Jersey needs that, otherwise Jersey build fails with: /src/main/java/org/glassfish/jersey/gf/ejb/internal/EjbComponentProvider.java:[138,58] cannot access org.jvnet.hk2.config.ConfigBeanProxy
[ERROR]   class file for org.jvnet.hk2.config.ConfigBeanProxy not found

Reply:
hk2-config 2.5.0 was in hk2 not in GF. Now that hk2-config has been moved to GF the version matches GF so that could explain the wiring issue.

The version of hk2-config used by jersey should not impact the OSGi import package for glassfish-api. We've discussed previously that glassfish-api should have a direct dependency on hk2-config.

The previous version of jersey did not have a dependency on hk2-config, how did the build pass before ?

Reply:
Previous version has a dependency on some ancient ejb container. When moving Jakarta, we tried to adopt the latest, but it comes with a dependency to config

Reply:
And that dependency does not come transitively ? Did you try to use scope=provided or optional=true ?

Reply:
It has provided dependency, yes.

Reply:
I don't see it: https://github.com/eclipse-ee4j/jersey/blob/f449fd6a7c1e3a7223c9b92fa022063785e3861c/containers/glassfish/jersey-gf-ejb/pom.xml#L77

Reply:
Here https://github.com/eclipse-ee4j/jersey/blob/f449fd6a7c1e3a7223c9b92fa022063785e3861c/containers/glassfish/jersey-gf-ejb/pom.xml#L46

Reply:
Thats what were talking about

Reply:
We add scope=provided and rebuild Jersey

Reply:
The link points at  jakarta.ejb, how is that related to org.jvnet.hk2.config ?

You are saying that you need hk2-config to satisfy the jersey build. However that dependency messes-up the GF build because of transitive dependency resolution.

If you'd make the hk2-config optional it would become invisible when depending on jersey-gf-ejb-container and thus would likely avoid the issues related to a bad version of hk2-config being resolved in GF.

Reply:
I said we have done it

Reply:
Can you provide a pointer to the pom with the change ?